pkg/console/controllers/route/controller.go

```diff
@@ -104,12 +109,23 @@ func (c *RouteSyncController) sync() error {
 		return nil
 	case operatorsv1.Removed:
 		klog.V(4).Infoln("console is in a removed state: deleting route")
-		return c.removeRoute()
+		if err = c.removeCustomRoute(); err != nil {
+			return err
+		}
+		return c.removeRoute(api.OpenShiftConsoleName)
 	default:
 		return fmt.Errorf("unknown state: %v", operatorConfig.Spec.ManagementState)
 	}
 
 	updatedOperatorConfig := operatorConfig.DeepCopy()
+
```

Comment:
This high up in the controller, I'm a bit inclined to always call both functions, but hide away the conditional details within them:

```golang
// just always call SyncCustom, but within the func return early if the custom route IS NOT being used:
//   if ! customRoute { return }
_, _, customErrReason, err := c.SyncCustomRoute(updatedOperatorConfig)
status.HandleProgressingOrDegraded(updatedOperatorConfig, "CustomRouteSync", customErrReason, err)
// and always call default, but return early if the custom route IS being used:
// the return early on this one is just the health check, like you have, since we are still creating the route.
// if customRoute { return }
_, _, defaultErrReason, err := c.SyncRoute(updatedOperatorConfig)
status.HandleProgressingOrDegraded(updatedOperatorConfig, "RouteSync", defaultErrReason, err)
status.SyncStatus(c.operatorConfigClient, updatedOperatorConfig)

return err;
```

Reply:
I'm also leaning towards

```golang
c.SyncDefaultRoute // adding default for future clarity
c.SyncCustomRoute
```